### ISSUE TITLE
Failling pod-to-external-fqdn-allow-google-cnp connectivity test error: curl: (28) Resolving timed out after 5000 milliseconds 

### ISSUE THREAD

**Issue by wapson** [cilium/cilium#18293]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Hey, On freshly created GKE cluster with installed Cilium via helm `pod-to-external-fqdn-allow-google-cnp` (pod is the part of the [Connectivity Test](https://raw.githubusercontent.com/cilium/cilium/v1.11/examples/kubernetes/connectivity-check/connectivity-check.yaml) ) readiness probe is failing due to `curl: (28) Resolving timed out after 5000 milliseconds` error.
```
helm install cilium cilium/cilium --version 1.11.0 \ --namespace kube-system  --set nodeinit.enabled=true  --set nodeinit.reconfigureKubelet=true  --set nodeinit.removeCbrBridge=true  --set nodeinit.restartPods=true  --set cni.binPath=/home/kubernetes/bin  --set gke.enabled=true  --set ipam.mode=kubernetes  --set ipv4NativeRoutingCIDR=$NATIVE_CIDR --set encryption.enabled=true --set encryption.nodeEncryption=false --set encryption=ipsec
```

1. Pod is present in `cilium endpoint list`
```
ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                                           IPv6   IPv4            STATUS   
           ENFORCEMENT        ENFORCEMENT 
3073       Disabled           Enabled           31591      k8s:io.cilium.k8s.policy.cluster=default                                     172.25.64.137   ready   
                                                           k8s:io.cilium.k8s.policy.serviceaccount=default                                                      
                                                           k8s:io.kubernetes.pod.namespace=cilium-test                                                          
                                                           k8s:name=pod-to-external-fqdn-allow-google-cnp
```

2. Output from hubble observer
```
 kubectl -n kube-system exec cilium-jhk76 -- hubble observe --since 50m --pod cilium-test/pod-to-external-fqdn-allow-google-cnp-6d8b6d59fc-7ts5j

...
Dec 16 20:08:10.947: cilium-test/pod-to-external-fqdn-allow-google-cnp-6d8b6d59fc-7ts5j:44717 -> kube-system/kube-dns-b4f5c58c7-9fgqr:53 L3-L4 REDIRECTED (UDP)
Dec 16 20:08:10.947: cilium-test/pod-to-external-fqdn-allow-google-cnp-6d8b6d59fc-7ts5j:44717 -> kube-system/kube-dns-b4f5c58c7-9fgqr:53 to-proxy FORWARDED (UDP)
Dec 16 20:08:10.947: cilium-test/pod-to-external-fqdn-allow-google-cnp-6d8b6d59fc-7ts5j:44717 -> kube-system/kube-dns-b4f5c58c7-9fgqr:53 to-proxy FORWARDED (UDP)
...
```

3. Kube-dns pods does not contain any logs regarding that.

4. The other two tests from Connectivity test which include CNP work fine (`pod-to-a-allowed-cnp` & `pod-to-a-denied-cnp` Cilium Network Policies)

5. Same request as `pod-to-external-fqdn-allow-google-cnp` from another pods is performing correctly

I also have two observations which may be related to this problem. 
1. [k8s-unmanaged.sh ](https://github.com/cilium/cilium/blob/master/contrib/k8s/k8s-unmanaged.sh) script returns that some pods (including 3 from connectivity test) are unmanaged by Cilium
2. Command from this paragraph about restarting unmanaged pods will restart basically all of the pods [restart-unmanaged-pods](https://docs.cilium.io/en/v1.9/gettingstarted/k8s-install-gke/#restart-unmanaged-pods) 

### Cilium Version

1.11.0

### Kernel Version

5.4.120+

### Kubernetes Version

v1.20.10-gke.1600

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aanm** [comment]
Hello, have you followed [this](https://docs.cilium.io/en/v1.11/gettingstarted/k8s-install-default/#install-cilium) guide and created your node pools with a node taint? Thank you.

**Comment by wapson** [comment]
Hey, thank you for you comment.
Due to some automation issues, I can't taint nodes with cilium taint. That's why I used `--set nodeinit.restartPods=true` in helm parameters to be sure that all resources will be restarted after Cilium installation. Of course in debugging sessions I rerolled all of the workloads a couple of times and the result is exactly the same.
Correct me if I'm wrong but as far as I understand tainting nodes is not needed if I will do that?
